### PR TITLE
Feat(serialization): AsBytes Refactor and Improvements

### LINKED ISSUE #744: Serialization -> AsBytes refactor and improvements
# Motivation

Serializable trait was created when we started the library to provide an interface to serialize types. Now that we have Serde support, the role is mostly covered by it. Still, there are times when we just want to convert some data to bytes, for example, to feed data to a transcript, so the functionality is still useful. 

# Proposal

- [ ] Change the `Serializable` trait to `AsBytes`, and `.serialize()` to `.as_bytes()`. This is more consistent with how other structures in Rust works, like Strings
- [ ] For all types that support `.as_bytes()`, support `into()` to create a `Vec::<u8>`
- [ ] For all types that support `.as_bytes()`, support `try_into()` to try to convert the data into an `[u8]`

### PR DESCRIPTION
**Pull request by PatStiles**
# AsBytes Refactor and Improvements

## Description

Partially Closes #744. 
 
- [x] Change the Serializable trait to AsBytes, and .serialize() to .as_bytes(). This is more consistent with how other [x] - structures in Rust works, like Strings
- [x] For all types that support .as_bytes(), support into() to create a Vec::<u8>
- [x] For all types that support .as_bytes(), support try_into() to try to convert the data into an [u8] (Derived from From implementation)

## Checklist
- [x] Linked to Github Issue

### CODE REVIEW

**Review by diegokingston — APPROVED** [review]

**Review by MauroToscano — APPROVED** [review]

**Comment by MauroToscano** [comment]
Pat, this should solves task1, right ?

File: math/src/unsigned_integer/element.rs [hunk 1]
```diff
@@ -962,6 +964,25 @@ impl<const NUM_LIMBS: usize> From<UnsignedInteger<NUM_LIMBS>> for u16 {
...
 
+#[cfg(feature = "alloc")]
+impl<const NUM_LIMBS: usize> AsBytes for UnsignedInteger<NUM_LIMBS> {
+    fn as_bytes(&self) -> alloc::vec::Vec<u8> {
+        self.limbs
+            .into_iter()
+            .fold(alloc::vec::Vec::new(), |mut acc, limb| {
+                acc.extend_from_slice(&limb.as_bytes());
+                acc
+            })
```
Comment (Oppen) [inline math/src/unsigned_integer/element.rs:975]:
```suggestion
            .flat_map(|limb| limb.as_bytes())
            .collect();
```

**Review by Oppen — APPROVED** [review]
LGTM. Left a little suggestion if you want to apply it, but it's good either way.

**Comment by PatStiles** [comment]
@MauroToscano From<_> implements into() as well.

**Pull request merged**